server_test.go

```diff
@@ -418,6 +418,20 @@ func TestEndToEnd(t *testing.T) {
 			// Wait for the goroutine to die.
 			<-ch
 		}
+
+		// validate when there is no entry in cache,
+		clearEntry(foo)
+		clearEntry(qux)
+		req.InitialResourceVersions = map[string]string{
+			foo: "0",
+		}
+		req.ResourceNamesSubscribe = []string{ads.WildcardSubscription}
+		stream, cancel = newStream(t)
```

Comment:
Interesting! I guess you must make sure to call `cancel` then :) 
![image](https://github.com/user-attachments/assets/743fca3a-b392-478b-8f29-8c6c94e0741b)